app/controllers/auth_controller.rb

```diff
@@ -0,0 +1,33 @@
+class AuthController < ApplicationController
+  def login
+    user = User.find_by(username: login_params[:username])
+    if user && user.authenticate(login_params[:password])
```

Comment:
Do you have to check for the existence of the params before accessing them?

Reply:
Not in this case, since the params hash is always present and we are only accessing 1 layer deep. If the key doesn't exist, `nil` will be returned, which is what we want.